rpcs3/Input/pad_thread.cpp

```diff
@@ -197,6 +200,27 @@ void pad_thread::Init()
 			{
 				input_log.notice("Pad %d: config=\n%s", i, cfg->to_string());
 			}
+
+			// If the user changes the class type of a controller, then send a disconnection notification using an unconfigured pad
+			if ((pad_settings[i].class_type != pad->m_class_type) && m_pads_connected[i])
```

Comment:
this is the wrong location to do this.
It should be done in update_pad_states

Reply:
Ive changed it to start a timer that gets handled in update_pad_states, is that okay?